src/activities/menu/Menu.qml

```diff
@@ -459,6 +500,143 @@ ActivityBase {
 
         }
 
+
+        Rectangle
+        {
+            id: searchbar
+            width: horizontal ? parent.width/2 : parent.width - (section.width+10)
+            height: horizontal ? parent.height/10 : parent.height/17
+            visible:  menuActivity.currentTag === "search" ? true :false
+            anchors
+            {
+               bottom : horizontal ? activitiesGrid.top : undefined
+               top: horizontal ? undefined : parent.top
+               left: horizontal ? undefined : section.right
+            }
+            anchors.topMargin: horizontal ? undefined : 4
+            anchors.bottomMargin: horizontal ? undefined : 4
+            x: horizontal ? (parent.width/2) - (searchbar.width/2) : 0
+            opacity: 0.5
+            radius: 10
+            border.width: 2
+            border.color: "black"
+            gradient: Gradient {
+
+               GradientStop { position: 0.3; color: "#000" }
+               GradientStop { position: 0.9; color: "#666" }
+               GradientStop { position: 1.0; color: "#AAA" }
+
+           }
+
+            TextField
+            {
+                id : input
+                anchors.fill: parent
+                text: ""
+                textColor: "black"
+                font.pixelSize: 28
+                font.bold: true
+                horizontalAlignment: TextInput.AlignHCenter
+                verticalAlignment: TextInput.AlignVCenter
+                font.family:GCSingletonFontLoader.fontLoader.name
+                activeFocusOnPress:  !ApplicationSettings.isVirtualKeyboard
+                onTextChanged:
+                {
+                    ActivityInfoTree.display(input.text)
+                }
+            }
+        }
+
+        VirtualKeyboard
+        {
+            id:keyboard
+            parent: background
+            property string search;
+            property var letter;
+            property int rows;
+            width: horizontal ? main.width :parent.width - (section.width +10)
```

Comment:
i have set the height here ,just before width 
height:((keyHeight + rowSpacing) * 3)+ margin